### PR TITLE
Ports Text Filtering [DNM]

### PR DESCRIPTION
**Pull request by Cactus2388**
No more REEing and YEETing.

:cl: FlattestGuitar ported by Cactus2388
tweak: REEing or Yeeting will give you a witty line and stop you from spreading your degeneracy.
/:cl:

Isn't tested, might break stuff, give me a minute to test it on a private server to see if it works, probably shouldn't but DNM until then.

### CODE REVIEW

File: code/__HELPERS/text_filters.dm [hunk 1]
```diff
@@ -0,0 +1,17 @@
+GLOBAL_DATUM_INIT(text_filter_datum, /datum/text_filter, new)
```
Comment (steamp0rt) [inline code/__HELPERS/text_filters.dm]: move this to `sunset/code/__HELPERS/text_filters.dm`

File: code/_globalvars/lists/names.dm [hunk 2]
```diff
@@ -26,6 +26,7 @@ GLOBAL_LIST_INIT(ing_verbs, world.file2list("strings/names/ing_verbs.txt"))
 GLOBAL_LIST_INIT(adverbs, world.file2list("strings/names/adverbs.txt"))
 GLOBAL_LIST_INIT(adjectives, world.file2list("strings/names/adjectives.txt"))
 GLOBAL_LIST_INIT(dream_strings, world.file2list("strings/dreamstrings.txt"))
+GLOBAL_LIST_INIT(retard_lines, world.file2list("strings/retard_lines.txt"))
```
Comment (steamp0rt) [inline code/_globalvars/lists/names.dm]: move this line to a new file `sunset/code/_globalvars/lists/names.dm`

File: code/modules/mob/living/say.dm [hunk 3]
```diff
@@ -154,6 +154,11 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	if(!can_speak_vocal(message))
 		to_chat(src, "<span class='warning'>You find yourself unable to speak!</span>")
 		return
+		
+	if(check_retard_speech(message))
```
Comment (steamp0rt) [inline code/modules/mob/living/say.dm]: Add `// sunset start` and `// sunset end`

File: code/modules/mob/mob_helpers.dm [hunk 4]
```diff
@@ -502,3 +502,9 @@ It's fairly easy to fix if dealing with single letters but not so much with comp
 
 /mob/proc/can_hear()
 	. = TRUE
+
+/proc/check_retard_speech(message = "")
```
Comment (steamp0rt) [inline code/modules/mob/mob_helpers.dm]: `sunset/code/modules/mob/mob_helpers.dm`

**Comment by steamp0rt** [comment]
Also please avoid using webeditor if you're changing more than 1 or 2 files.

**Comment by steamp0rt** [comment]
Also, add ligma to the filter while you're at it?

File: code/modules/mob/living/say.dm (hunk 3 above)
Comment (ComradeCheekiBreeki) [reply code/modules/mob/living/say.dm]: I'm assuming since these are comments they are for organization and won't change anything?

**Comment by ComradeCheekiBreeki** [comment]
Well first of all we have no checks, we kinda need those before we can start merging shit.

File: code/modules/mob/living/say.dm (hunk 3 above)
Comment (steamp0rt) [reply code/modules/mob/living/say.dm]: Yeah. Also help with merge conflicts.

Comment (ComradeCheekiBreeki) [reply code/modules/mob/living/say.dm]: Makes sense

Comment (steamp0rt) [reply code/modules/mob/living/say.dm]: if a file conflicts, I can check for our stuff just by Ctrl+F'ing for "sunset".

File: code/__HELPERS/text_filters.dm (hunk 1 above)
Comment (Cactus2388) [reply code/__HELPERS/text_filters.dm]: I'm pretty sure that's where it already is.

File: code/_globalvars/lists/names.dm (hunk 2 above)
Comment (Cactus2388) [reply code/_globalvars/lists/names.dm]: It's already in `code/_globalvars/lists/names.dm` (please tell me if I'm being dumb and missed something)

File: code/__HELPERS/text_filters.dm (hunk 1 above)
Comment (steamp0rt) [reply code/__HELPERS/text_filters.dm]: No, it needs to be in `sunset/code/__HELPERS/text_filters.dm`

Notice the `sunset`?

Comment (ComradeCheekiBreeki) [reply code/__HELPERS/text_filters.dm]: It's already in `sunset/code/__HELPERS/text_filters.dm`

File: code/_globalvars/lists/names.dm (hunk 2 above)
Comment (steamp0rt) [reply code/_globalvars/lists/names.dm]: Needs to be in `sunset/code/_globalvars/lists/names.dm`

Note the `sunset`.

File: code/__HELPERS/text_filters.dm (hunk 1 above)
Comment (steamp0rt) [reply code/__HELPERS/text_filters.dm]: No it isn't, it's just in normal code folder, not sunset code.

Comment (ComradeCheekiBreeki) [reply code/__HELPERS/text_filters.dm]: Oh I see, I gotcha

Comment (ComradeCheekiBreeki) [reply code/__HELPERS/text_filters.dm]: God damnit my fucking monitor cord was obscuring the typepath >:(

File: code/modules/mob/living/say.dm (hunk 3 above)
Comment (ComradeCheekiBreeki) [reply code/modules/mob/living/say.dm]: Yeah I gotcha

File: code/__HELPERS/text_filters.dm (hunk 1 above)
Comment (steamp0rt) [reply code/__HELPERS/text_filters.dm]: 😆

**Comment by ComradeCheekiBreeki** [comment]
Move all of your files from `code/*` `to sunset/code/*`

**Comment by steamp0rt** [comment]
git pull the branch _sigh_

**Comment by SunsetStation** [comment]
this is still broken

**Comment by steamp0rt** [comment]
WTF? It wasn't even merged? I can't find it's files!

**Pull request merged**